docs/transpile/defaults_and_configuration_options.mdx

```diff
@@ -0,0 +1,150 @@
+- layout_method (str)
+- optimization_level (int)
+- optimization_method (str)
+- output_name (str | List[str])
+- routing_method (str)
+- scheduling_method (str)
+- seed_transpiler (int)
+- target (Target)
+- timing_constraints (Dict[str, int]) –
+- translation_method (str) 
+- unitary_synthesis_method (str)
+- unitary_synthesis_plugin_config (dict)
+
+## Commonly used parameters 
+ 
+### Optimization level
+
+Set the optimization level
+
+Decomposing quantum circuits into the basis gate set of the target device and the addition of swap gates needed to match hardware topology cause an increase the depth and gate count of quantum circuits. To mitigate this increased complexity, you can set the `optimization_level`.  Setting this value calls an optmization routine that optimizes circuits by combining or eliminating gates. In some cases these methods are so effective the output circuits have lower depth than the inputs. In other cases, not much can be done, and the computation may be difficult to perform on noisy devices. 
+
+Different gate optimizations are turned on with different `optimization_level` values. In general, the higher the number, the more optimized it is, at the expense of longer processing time. Choosing the best optimization level might take trial and error, as it depends heavily on the circuit being transpiled and the backend being targeted. By default, `optimization_level=1` is used.
+
+Levels: 
+
+
+The `optimization_level` setting specifies how much optimization to perform on the circuits. Higher levels generate more optimized circuits, at the expense of longer compile times.
+
+- `optimization_level=0`: Trivial optimization, which maps the circuit to the backend with no explicit optimization (except whatever optimizations the mapper does).
+- `optimization_level=1`: Light optimization, which maps the circuit and does simple optimizations by collapsing adjacent gates. This is the default value.
+- `optimization_level=2`: Medium optimization, including a noise-adaptive layout and a gate-cancellation procedure based on gate commutation relationships.
+- `optimization_level=3`: Heavy optimization, which performs the optimization included in levels 1 - 2 and resynthesizes two-qubit gate blocks in the circuit.
+
+<Table>
+  <thead>
+    <Tr>
+      <Th>Optimization Level</Th>
+      <Th>Description</Th>
+    </Tr>
+  </thead>
+  <tbody>
+    <Tr>
+      <Td>0</Td>
+      <Td>
+        No optimization: typically used for hardware characterization
+        - Basic translation
+        - Layout (as specified)
+        - Routing (stochastic swaps)
+      </Td>
+    </Tr>
+    <Tr>
+      <Td>1</Td>
+      <Td>
+        Light optimization (default):
+
+        -   Layout (trivial → vf2 → SabreLayout if routing is required)
```

Comment:
One thing I think is super confusing is the difference between Layout and Routing (together I call these Mapping, but that might be a personal thing). At a functional level, I think of them as

- Routing: insert SWAPs due to the limited connectivity of the lattice
- Layout: pick the set of qubits in the lattice which to use based on their noise properties

However, in practice these seem to be somewhat conflated, likely do to needing to pick a set of qubits before you minimize the number of SWAPs? And I think of Sabre as being used for Routing and VF2 as used for Layout. Is this the right place to discuss this, or should be be linking to another doc where this is discussed.

Reply:
Tl;dr It might be worth saving this for some of the details on stages section of the docs, because this makes more sense there talking about how all the stages are constructed. For the definition of what's in the different optimization levels by default even the list of trivial -> vf2 -> sabre might be too much detail for this document.

Yeah, they're tightly coupled. Sabre is actually both a routing algorithm and a layout algorithm (it selects a layout by using a routing algorithm but instead of inserting swap gates it swaps the qubits virtually in the layout). The typical pipeline outlined here for level 1 is we first try the trivial layout (0->0, 1->1, etc) if that matches device connectivity we use it and move on (skipping routing). If it doesn't then we run `VF2Layout` (which is different from `VF2PostLayout`) which uses rustworkx to find an isomorphic subgraph on the device to execute the circuit on with no swaps (and if it finds > 1 select the 1 with best noise properties) and also skip routing, then if nothing can be found we run `SabreLayout` to select a good heuristic layout.

Just to make it extra confusing, `SabreLayout` also runs routing and inserts swaps. This is done as an efficiency because the whole algorithm is written in rust and it's much faster to compute the layout, find the swaps with that layout, and then select the layout which gives you the best output and just return that as a single thing. Then to make it even more confusing `VF2PostLayout` runs as part of the routing stage to pick a better layout because it only can be used after routing. But these are implementation details of the specific algorithms we use in the presets and the stages are logically distinct but related at a high level (i.e. you need a layout to run routing because you can't compute where to insert swaps until you know which physical qubits you're running on).

Reply:
Added this discussion to the issue for details about transpiler stages: https://github.com/Qiskit/documentation/issues/104#issuecomment-1790981912